### ISSUE TITLE
[Bug] Reports 0 tests run when specs are in third level of suite hierarchy

### ISSUE THREAD

**Issue by larmitage-bjss** [daun/playwright-report-summary#147]
I have two projects generating playwright json files, one has the spec details inside the second level of suites and the summary is working correctly:

<img width="599" alt="Screenshot 2024-04-17 at 17 22 32" src="https://github.com/daun/playwright-report-summary/assets/156074260/01149a20-71fa-4106-afd4-6dedb569bb14">

The second has the spec details inside the third level of suites and the summary is coming out as "0 tests across 4 suites":

<img width="505" alt="Screenshot 2024-04-17 at 17 22 40" src="https://github.com/daun/playwright-report-summary/assets/156074260/f73b3716-5dd2-4a89-b41b-727047fd3620">

I can see the issue is in [report.ts](https://github.com/daun/playwright-report-summary/blob/main/src/report.ts#L157) where it is only looking at the first and second level of suites, but I have been struggling to update the code as the tests are currently not working.

It should just need updating to something like:
```
const specs: SpecSummary[] = report.suites.reduce((all, file) => {
	for (const spec of file.specs) {
		all.push(parseSpec(spec, [file]))
	}
	for (const suite of file.suites || []) {
		for (const spec of suite.specs) {
			all.push(parseSpec(spec, [file, suite]))
		}

		for (const suite2 of suite.suites || []) {
			for (const spec of suite2.specs) {
				all.push(parseSpec(spec, [file, suite, suite2]))
			}
		}
	}
	return all
}, [] as SpecSummary[])
```

Or even better a recursive function until it finds an empty array of suites.

Thanks

**Comment by daun** [comment]
Makes sense. I never knew one could just recursively nest suites! If you can provide an example json report that's failing to generate the correct summary, I'm happy to take a look at it.

**Comment by daun** [comment]
I think the tests are currently failing because of too strict an eslint or typescript config, not because the logic is faulty.

**Comment by larmitage-bjss** [comment]
I can share examples but I'd rather not do it publicly. I've just followed you on Twitter/X, would you be able to follow me back and I can DM you? Unless you've got a different method (public email address etc)?

**Comment by daun** [comment]
@larmitage-bjss I've sent you a dm, let me know if that works.

**Comment by daun** [comment]
I've released 3.3.0 which should take care of this. Can you try it out and report back if it's working for you?

**Comment by larmitage-bjss** [comment]
Absolute star, looks like it's working correctly now! Thank you so much!

![Screenshot 2024-04-22 at 09 49 58](https://github.com/daun/playwright-report-summary/assets/156074260/d37bc081-e802-48f6-9749-e78367880c3e)
